CHANGELOG.md

```diff
@@ -12,6 +12,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Replace latches with CompletableFutures for extensions ([#5646](https://github.com/opensearch-project/OpenSearch/pull/5646))
 - Add support to disallow search request with preference parameter with strict weighted shard routing([#5874](https://github.com/opensearch-project/OpenSearch/pull/5874))
 - Added support to apply index create block ([#4603](https://github.com/opensearch-project/OpenSearch/issues/4603))
+- Hardened token permissions in GitHub workflows ([#4587](https://github.com/opensearch-project/OpenSearch/pull/4587))
```

Comment:
Build-related stuff can stay out of the CHANGELOG so let's nix this. We can also remove the line from the 3.0 section of the CHANGELOG on the main branch.

Reply:
Removed.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/6098